docs/post-mortems/2019-01-23.md

```diff
@@ -0,0 +1,95 @@
+# January 23rd, 2019 Prow Outage
+
+Created By: Benjamin Elder <bentheelder@google.com>
+Last Modified: 2019-01-23
+
+## Summary
+
+The main Kubernetes production Prow deployment, [prow.k8s.io] stopped responding
+to GitHub events due to the webhook handler ("[hook]") being down.
+
+This resulted in an outage of unknown length bounded to at maximum <compute and fill in>
+
+## Impact
+
+Prow did not respond to any github events, including our various slash commands,
+test triggering due to PR changes, etc.
+
+Already scheduled, and time-based scheduled test running continued, 
+but using config prior to merging the non-validating config.
+
+## Root Cause
+
+When the job config was [automatically updated] from a [preset refactor PR] 
+components failed to load the new configuration due to errors like:
+
+> {"component":"hook","error":"duplicated preset label : preset-aws-credential","level":"fatal","msg":"Error starting config agent.","time":"2019-01-23T16:08:30Z"}
+
+This is because the PR refactored some job configuration presets to use
+a common label key with different label values. The config "agent" validation
+rejected this format, but for some reason this was not caught in presubmit.
```

Comment:
There was a presubmit unit test which checked for uniqueness of preset key. I changed it in this PR https://github.com/kubernetes/test-infra/pull/10868/files to check uniqueness of 'key:value' instead. So that we're able to use the same label key with different values (like how k8s uses labels for categorizing pods).

Reply:
Yes. I failed to identify the combination of prow change + config change. This is why we should slow down and not rush sweeping refactors...

Reply:
Those changes needed to be separate PRs, with config changed happening after the next prow upgrade.

Reply:
Agree. Sorry, I didn't know though that change wasn't deployed automatically.

Reply:
Hummmm - I thought Prow should use the last known good config, if the new config is bad?

Reply:
It did for all components except hook, however this is still bad because not using new config is potentially a form of outage.